runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1199,18 +1199,19 @@ function M.show_line_diagnostics(opts, bufnr, line_nr, client_id)
   end
 
   local line_diagnostics = M.get_line_diagnostics(bufnr, line_nr, opts, client_id)
-  if vim.tbl_isempty(line_diagnostics) then return end
+  local num_diagnostics = vim.tbl_count(line_diagnostics)
+  if num_diagnostics == 0 then return end
 
   for i, diagnostic in ipairs(line_diagnostics) do
-    local prefix = string.format("%d. ", i)
+    local prefix = num_diagnostics == 1 and '' or string.format("%d. ", i)
```

Comment:
Should we do a `'%0' .. tostring(num_diagnostics) .. '%d'` here to ensure all listed items are aligned the same if there are more than 9, or more than 99 diagnostics?